services/base-json.spec.js

```diff
@@ -35,44 +35,44 @@ class DummyJsonService extends BaseJsonService {
 
 describe('BaseJsonService', function() {
   describe('Making requests', function() {
-    let sendAndCacheRequest, serviceInstance
+    let sendAndCacheRequest
     beforeEach(function() {
       sendAndCacheRequest = sinon.stub().returns(
         Promise.resolve({
           buffer: '{"some": "json"}',
           res: { statusCode: 200 },
         })
       )
-      serviceInstance = new DummyJsonService(
-        { sendAndCacheRequest },
-        { handleInternalErrors: false }
-      )
     })
 
     it('invokes _sendAndCacheRequest', async function() {
-      await serviceInstance.invokeHandler({}, {})
+      await DummyJsonService.invoke(
+        { sendAndCacheRequest },
+        { handleInternalErrors: false }
+      )
 
       expect(sendAndCacheRequest).to.have.been.calledOnceWith(
         'http://example.com/foo.json',
-        {
-          headers: { Accept: 'application/json' },
-        }
+        { headers: { Accept: 'application/json' } }
       )
     })
 
     it('forwards options to _sendAndCacheRequest', async function() {
-      Object.assign(serviceInstance, {
+      class WithOptions extends DummyJsonService {
```

Comment:
The only loss due to the above is that we have to build a new class; we can't modify the service instance.